cmd/containerd-shim-runhcs-v1/serve.go

```diff
@@ -169,23 +169,23 @@ var serveCommand = cli.Command{
 
 		ttrpcAddress := os.Getenv(ttrpcAddressEnv)
 		ttrpcEventPublisher, err := newEventPublisher(ttrpcAddress)
-
 		if err != nil {
 			return err
 		}
-
 		defer func() {
 			if err != nil {
 				ttrpcEventPublisher.close()
 			}
 		}()
 
 		// Setup the ttrpc server
-		svc = &service{
-			events:    ttrpcEventPublisher,
-			tid:       idFlag,
-			isSandbox: ctx.Bool("is-sandbox"),
+		svc, err = NewService(WithEventPublisher(ttrpcEventPublisher),
+			WithTID(idFlag),
+			WithIsSandbox(ctx.Bool("is-sandbox")))
+		if err != nil {
+			return fmt.Errorf("starting service: %w", err)
```

Comment:
I'd prefer to have "failed to start (or create new I guess in this case) service: %w"